src/diffusers/models/modeling_utils.py

```diff
@@ -859,6 +976,51 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
 
         return model
 
+    # Taken from `transformers`.
+    @wraps(torch.nn.Module.cuda)
+    def cuda(self, *args, **kwargs):
+        # Checks if the model has been loaded in 8-bit
+        if getattr(self, "quantization_method", None) == QuantizationMethod.BITS_AND_BYTES:
+            raise ValueError(
+                "Calling `cuda()` is not supported for `4-bit` or `8-bit` quantized models. Please use the model as it is, since the"
+                " model has already been set to the correct devices and cast to the correct `dtype`."
+            )
+        else:
+            return super().cuda(*args, **kwargs)
+
+    # Taken from `transformers`.
+    @wraps(torch.nn.Module.to)
+    def to(self, *args, **kwargs):
+        # Checks if the model has been loaded in 8-bit
+        if getattr(self, "quantization_method", None) == QuantizationMethod.BITS_AND_BYTES:
+            raise ValueError(
+                "`.to` is not supported for `4-bit` or `8-bit` bitsandbytes models. Please use the model as it is, since the"
+                " model has already been set to the correct devices and cast to the correct `dtype`."
+            )
+        return super().to(*args, **kwargs)
```

Comment:
We recently merged this PR: https://github.com/huggingface/transformers/pull/33122 to remove the to() and cuda() restriction on 4-bit models.

Reply:
Resolved in 31725aa23